Core/WebSocket.m

```diff
@@ -201,7 +201,7 @@ - (id)delegate
 	__block id result = nil;
 	
 	dispatch_sync(websocketQueue, ^{
-		result = delegate;
+		result = self.delegate;
```

Comment:
The main question for all of these `self` additions in dispatch/blocks. Usually, when I access `self` in a `dispatch/block` I make it a `__block` variable outside and then call that from within the block so that the reference count doesn't increment (at least that is what I thought it did).
Example:
`   __block typeof(self) bSelf = self;`

Reply:
Yeah, that might be the more correct method, I just took the "fix it" suggestion from Xcode.